Libraries/oneMKL/matrix_mul_mkl/matrix_mul_mkl.cpp

```diff
@@ -57,7 +52,36 @@ void test(queue &Q, int M, int N, int K)
         return duration<double>(end - start).count();
     };
 
-    /* Do a warmup call to initialize MKL and ensure kernels are JIT'ed if needed */
+    /* Fill A/B with all ones to verify correctness */
+    generate_ones(rd_size, host_data);
+    replicate_data(Q, A, lda * K, host_data, rd_size);
+    replicate_data(Q, B, ldb * N, host_data, rd_size);
+
+    /* Verify that the leading entries of C are correct */
+    std::cout << " -> Verification...";
+    (void) time_gemms(1);
+    size_t elems = std::min(ldc * N, rd_size);
+    Q.copy(C, host_data, elems).wait();
+    bool ok = true;
+    int linear_id = 0;
+    for (size_t j = 0; j < N; j++) {
+        for (size_t i = 0; i < M; i++) {
+            linear_id = j*ldc + i;
+            if (linear_id >= elems) break;
+            if (host_data[linear_id] != T(K)) {
+                ok = false;
+            }
+        }
+        if (linear_id >= elems) break;
+    }
+    std::cout << (ok ? " passes." : " FAILS!") << std::endl;
```

Comment:
Could you please also add `exit(1)` in case of test's failure like it's done for other oneMKL samples? It's a criteria for the QA team to understand whether the test fails or not

Reply:
Made a follow-up PR #2517 to fix this.